packages/abi-coder/package.json

```diff
@@ -17,6 +17,7 @@
     "dist"
   ],
   "scripts": {
+    "pretest": "tsc --noEmit",
```

Comment:
Why is this necessary?

Reply:
Because this is how the testing of `*.compilation-test.ts` files is happening. If those files compile with tsc, the type inference works as expected.